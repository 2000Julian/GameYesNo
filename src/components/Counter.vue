
<template>
  <h2>{{ customTitle }} </h2>
  <p data-testid="counter">{{ counter}}</p>
  <p>{{ counter }} <sup>9</sup> = {{ squareCounter }}</p>
  <div class="class- mhv">
    <button v-on:click="increment()">+1</button>
    <button data-testid="decrease" v-on:click="decrease()">-1</button>
  </div>
</template>





<script>
export default {
    props: {
        title: String,
        start: {
            type: Number,
            default: 100,
            validator( value ) {
                return value >=0
            }
           // required: true
        },
    },
  // name: 'patito'
  data() {
    //la data es una funcion que tiene que regresar un objeto, //la data se crea el objeto que definimos de una manera reactiva
    return {
      //lo que este dentro de aqui es reactivo, que significa que vue estara pendiente de cualquier cambio que sufra y reaccionara acorde
      counter: this.start,
    };
  },
  methods: {
    getSquareValue() {
      return this.counter * this.counter;
    },
    decrease() {
      this.counter --;
    },
    increment(){
        this.counter = this.counter + 1;
    }

  },
  computed: {
    squareCounter() {
      return this.counter * this.counter;
    },
    customTitle() {
        return this.title || 'Counter'
    }
  },
};
</script>




<style>
button {
  background-color: #64b687;
  border-radius: 5px;
  border: 1px solid white;
  color: white;
  margin: 0 5px;
  padding: 5px 15px;
  transition: 0.3s ease-in-out;
}

button:hover {
 background-color: #5aa ;
 transition: 0.3s ease-in-out;
}


</style>
