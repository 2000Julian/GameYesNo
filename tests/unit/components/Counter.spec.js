import {shallowMount} from '@vue/test-utils'
import Counter  from '@/components/Counter'



describe('Counter component', ()=>{

    let wrapper 

    beforeEach(()=>{
        wrapper = shallowMount ( Counter )
    })



   // test('Debe de ser hacer math con el snapchop', ()=>{

    //         //wrapper estandar que envuelve todo
    //     const wrapper = shallowMount( Counter )
    //     expect( wrapper.html() ).toMatchSnapshot()//captura una copia de codigo htmal y lo compara a ver si cambio
    // })
     test('h2 debe tener el valor por defecto "Counter', () =>{
        expect( wrapper.find('h2').exists() ).toBeTruthy()//verifica que exista elelemento
        const h2=  wrapper.find('h2')

        console.log( h2.text() )
        expect (h2.text() ).toBe( 'Counter')
     })

     test('El valor por defecto debe ser 100 en el p',async ()=>{

        //const pTags = wrapper.findAll('p')
        const value = wrapper.find('[data-testid="counter"]').text()
        console.log( value )
        //expect( value[1].text() ).toBe('100')
        expect( value ).toBe('100')

     })

     test('Debe de incrementar y decrementar el valor del contador', async() =>{

         const [increase, decrease]  = wrapper.findAll('button')

        await increase.trigger('click')
        await increase.trigger('click')
        await increase.trigger('click')
        await decrease.trigger('click')
        await decrease.trigger('click')
         //expect( value ).toBe('101')
         //TODO: Tarea
        const value = wrapper.find('[data-testid="counter"]').text()
        expect( value ).toBe('101')
     })

     test('Debe de establecer el valor por defecto', () =>{

        const { start } = wrapper.props()
        console.log(start)

        const value = wrapper.find('[data-testid="counter"]').text()

        expect( Number( value )).toBe( start )
     })


     test('Debe de mostrar la prop title', () =>{
        const title = 'Hola Mundo!!!'
        const wrapper = shallowMount ( Counter, {
            props: {
                title,
                start: '5'
            }
        } )

        const { start } = wrapper.props()
        console.log( typeof start )

        console.log( wrapper.html() )

        expect( wrapper.find( 'h2' ).text()).toBe( title )
     })
})