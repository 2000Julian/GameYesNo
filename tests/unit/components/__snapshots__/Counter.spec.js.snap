// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Debe de ser hacer math con el snapchop 1`] = `
<h2>Countrer</h2>
<p>100 <sup>9</sup> = 10000</p>
<div class="class- mhv"><button>+1</button><button>-1</button></div>
`;
